cql3/restrictions/statement_restrictions.hh

```diff
@@ -414,7 +414,11 @@ public:
     }
 
     bool ck_restrictions_need_filtering() const {
-        return _partition_key_restrictions->has_unrestricted_components(*_schema) || _clustering_columns_restrictions->needs_filtering(*_schema);
+        return _partition_key_restrictions->has_unrestricted_components(*_schema) 
+            // In TOKEN restrictions on tables with secondary indexes we only restrict idx_token column in view,
+            // leaving all partition key columns unrestricted.
```

Comment:
I'm having trouble understanding this.  Are we fetching too many rows from the index table?  That seems odd, given that TOKEN is the first clustering column there.

Reply:
Example that shows this case:
```
CREATE TABLE ks.t(pk1 int, ck1 int, ck2 int, v int, PRIMARY KEY(pk1, ck1, ck2));
CREATE INDEX ON ks.t(v);
SELECT * FROM ks.t WHERE v = 5 AND token(pk1) < 0 AND ck1 = 1 AND ck2 = 2;
```

Created global index has primary key: `((v), idx_token, pk1, ck1, ck2)`.

When doing the index query, we have the following partition slice: `idx_token < 0` (partition: `v = 5`). We cannot "add" `ck1` and `ck2` to this slice, because there is `pk1` in between `idx_token` and `ck1`, and we don't know the value of `pk1`. 

Therefore, we have to do some filtering on `ck1` and `ck2`. But previously `ck_restrictions_need_filtering()` returned `false`, because: `_partition_key_restrictions->has_unrestricted_components(*_schema)` was `false` - `TOKEN(pk1)` restriction and `_clustering_columns_restrictions->needs_filtering(*_schema)` was `false` - whole clustering key is restricted.

Reply:
Updated the comment in the amended commit.